forms-flow-components/src/components/CustomComponents/FormBuilderModal.tsx

```diff
@@ -0,0 +1,152 @@
+  secondoryBtndataTestid?: string;
+  primaryBtnariaLabel?: string;
+  secondoryBtnariaLabel?: string;
+  closedataTestid?: string;
+  nameInputDataTestid?: string;
+  descriptionDataTestid?: string;
+ 
+}
+
+export const FormBuilderModal: React.FC<BuildFormModalProps> = React.memo(
+  ({
+    showBuildForm,
+    onClose,
+    handleChange,
+    primaryBtnAction,
+    secondaryBtnAction = onClose, // Default to onClose if not provided
+    setFormDescription,
+    setNameError,
+    nameError,
+    formSubmitted,
+    modalHeader,
+    nameLabel = "Name",
+    descriptionLabel = "Form Description",
+    primaryBtnLabel = "Save and Edit form",
+    secondaryBtnLabel = "Cancel",
+    nameValidationOnBlur,
+    primaryBtndataTestid = "confirm-button",
+    secondoryBtndataTestid = "cancel-button",
+    primaryBtnariaLabel = "Create and Edit form",
+    secondoryBtnariaLabel = "Cancel",
+    closedataTestid = "close",
+    nameInputDataTestid = "form-name",
+    descriptionDataTestid = "form-description",
+    
+  }) => {
+    const textareaRef = useRef<HTMLTextAreaElement | null>(null);
+
+    useEffect(() => {
+      const textarea = textareaRef.current;
+      if (textarea) {
+        const handleInput = () => {
+          textarea.style.height = "auto";
+          textarea.style.height = `${textarea.scrollHeight}px`;
+        };
+        textarea.addEventListener("input", handleInput);
+        return () => {
+          textarea.removeEventListener("input", handleInput);
+        };
+      }
+    }, [showBuildForm]);
+
+    return (
+      <>
+        <Modal
+          show={showBuildForm}
+          onHide={onClose}
+          dialogClassName="modal-50w"
+        >
+          <Modal.Header>
+            <Modal.Title>
+              <b>{modalHeader}</b>
+            </Modal.Title>
+            <div className="d-flex align-items-center">
+            <CloseIcon onClick={onClose} />
+            </div>
+          </Modal.Header>
+          <Modal.Body className="p-5">
+            <label className="form-label">{nameLabel}</label>
+            <span className="valiation-astrisk">*</span>
+            <input
+              type="text"
+              placeholder="Form 1"
+              className={`form-input ${nameError ? "input-error" : ""}`}
+              aria-label="Name of the form"
+              data-testid={nameInputDataTestid}
+              onBlur={nameValidationOnBlur}
+              onChange={(event) => {
+                setNameError("");
+                handleChange("title", event);
+              }}
+              required
+            />
+            {nameError && (
+              <div className="validation-text mb-4">{nameError}</div>
+            )}
+
+            <label className="form-label">{descriptionLabel}</label>
+
+            <FormTextArea
+              onChange={(e) => setFormDescription(e.target.value)}
+              ref={textareaRef}
+              placeholder="Short description of the form"
```

Comment:
please replace it with placeholder={placeholderForDescription}

Reply:
Resolved